src/bitcoin.cpp

```diff
@@ -0,0 +1,287 @@
+            // Add escaping to the command line that the executable being
+            // invoked will split up using the CommandLineToArgvW function,
+            // which expects arguments with spaces to be quoted, quote
+            // characters to be backslash-escaped, and backslashes to also be
+            // backslash-escaped, but only if they precede a quote character.
+            std::string escaped{'"'}; // Start with a quote
+            for (size_t i = 0; i < arg.size(); ++i) {
+                if (arg[i] == '\\') {
+                    // Count consecutive backslashes
+                    size_t backslash_count = 0;
+                    while (i < arg.size() && arg[i] == '\\') {
+                        ++backslash_count;
+                        ++i;
+                    }
+                    if (i < arg.size() && arg[i] == '"') {
+                        // Backslashes before a quote need to be doubled
+                        escaped.append(backslash_count * 2 + 1, '\\');
+                        escaped.push_back('"');
+                    } else {
+                        // Otherwise, backslashes remain as-is
+                        escaped.append(backslash_count, '\\');
+                        --i; // Compensate for the outer loop's increment
+                    }
+                } else if (arg[i] == '"') {
+                    // Escape double quotes with a backslash
+                    escaped.push_back('\\');
+                    escaped.push_back('"');
+                } else {
+                    escaped.push_back(arg[i]);
+                }
+            }
+            escaped.push_back('"'); // End with a quote
+            escaped_args.emplace_back(std::move(escaped));
+            new_argv.push_back((char *)escaped_args.back().c_str());
+        }
+    }
+    new_argv.push_back(nullptr);
+    return _execvp(file, new_argv.data());
+#endif
+}
+
+//! Execute the specified bitcoind, bitcoin-qt or other command line in `args`
+//! using src, bin and libexec directory paths relative to this executable, where
+//! the path to this executable is specified in `wrapper_argv0`.
+//!
+//! @param args Command line arguments to execute, where first argument should
+//!             be a relative path to a bitcoind, bitcoin-qt or other executable
+//!             that will be located on the PATH or relative to this executable.
+//!
+//! @param wrapper_argv0 String containing first command line argument passed to
+//!                      main() to run the current executable. This is used to
+//!                      help determine the path to the current executable and
+//!                      how to look for new executables.
+//
+//! @note This function doesn't currently print anything but can be debugged
+//! from the command line using strace or dtrace like:
+//!
+//!     strace -e trace=execve -s 10000 build/src/bitcoin ...
+//!     dtrace -n 'proc:::exec-success { trace(curpsinfo->pr_psargs); }' -c ...
```

Comment:
The `dtrace` example would print all exec calls on the system, not just the ones by the given program.

```suggestion
//!     dtrace -n 'proc:::exec-success /pid == $target/ { trace(curpsinfo->pr_psargs); }' -c ...
```